wkcuber/api/dataset.py

```diff
@@ -647,81 +586,33 @@ def get_or_create(
         else:
             return cls.create(dataset_path, scale, name)
 
-    def _create_layer(
-        self,
-        layer_name: str,
-        dtype_per_channel: np.dtype,
-        num_channels: int,
-        category: str,
-    ) -> Layer:
-        layer_type = (
-            Layer if category == LayerCategories.COLOR_TYPE else SegmentationLayer
-        )
-        return layer_type(layer_name, self, dtype_per_channel, num_channels)
-
     def set_view_configuration(
-        self, view_configuration: "DatasetViewConfiguration"
+        self, view_configuration: DatasetViewConfiguration
     ) -> None:
-        self.properties._default_view_configuration = {
-            _snake_to_camel_case(k): v
-            for k, v in vars(view_configuration).items()
-            if v is not None
-        }
-        self.properties._export_as_json()  # update properties on disk
-
-    def get_view_configuration(self) -> Optional["DatasetViewConfiguration"]:
-        view_configuration_dict = self.properties.default_view_configuration
-        if view_configuration_dict is None:
-            return None
-
-        return DatasetViewConfiguration(
-            four_bit=view_configuration_dict.get("fourBit"),
-            interpolation=view_configuration_dict.get("interpolation"),
-            render_missing_data_black=view_configuration_dict.get(
-                "renderMissingDataBlack"
-            ),
-            loading_strategy=view_configuration_dict.get("loadingStrategy"),
-            segmentation_pattern_opacity=view_configuration_dict.get(
-                "segmentationPatternOpacity"
-            ),
-            zoom=view_configuration_dict.get("zoom"),
-            position=cast(
-                Tuple[int, int, int], tuple(view_configuration_dict["position"])
-            )
-            if "position" in view_configuration_dict
-            else None,
-            rotation=cast(
-                Tuple[int, int, int], tuple(view_configuration_dict["rotation"])
-            )
-            if "rotation" in view_configuration_dict
-            else None,
-        )
+        self._properties.default_view_configuration = view_configuration
+        self._export_as_json()  # update properties on disk
+
+    def get_view_configuration(self) -> Optional[DatasetViewConfiguration]:
+        return self._properties.default_view_configuration
 
     def __repr__(self) -> str:
         return repr("Dataset(%s)" % self.path)
 
+    def _export_as_json(self) -> None:
+        with open(self.path / PROPERTIES_FILE_NAME, "w") as outfile:
+            json.dump(
+                dataset_converter.unstructure(self._properties),
+                outfile,
+                indent=4,
+                separators=(",", ": "),
```

Comment:
Those are the default seperators with indent anyways:
https://docs.python.org/3/library/json.html#json.dump
```suggestion
```